src/windows/win_broker_os_interface.cpp

```diff
@@ -0,0 +1,177 @@
+/******************************************************************************
+ * Copyright 2019 ETC Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************
+ * This file is a part of RDMnetBroker. For more information, go to:
+ * https://github.com/ETCLabs/RDMnetBroker
+ *****************************************************************************/
+
+#include "win_broker_os_interface.h"
+
+#include <iostream>
+#include <memory>
+#include <Windows.h>
+#include <ShlObj.h>
+
+constexpr const WCHAR kRelativeConfFileName[] = L"\\ETC\\RDMnetBroker\\broker.conf";
+static const std::vector<std::wstring> kRelativeLogFilePath = {L"ETC", L"RDMnetBroker"};
+static const std::wstring kLogFileName = L"broker.log";
+
+std::unique_ptr<char[]> ConvertPathToUtf8(const std::wstring& conf_file_path)
```

Comment:
Why not have this return a std::string?

Reply:
There was, at some point, a reason. Let me see if I can figure out what it was. 😄

Reply:
The reason is, alas, lost to time. A `std::string` it shall be.